src/error.rs

```diff
@@ -158,3 +161,24 @@ pub struct PalletError {
     /// The error description.
     pub description: Vec<String>,
 }
+
+/// Transaction error.
+#[derive(Clone, Debug, Eq, Error, PartialEq)]
+pub enum TransactionError {
+    /// The finality subscription expired
```

Comment:
State what the timeout is

Reply:
So, subxt has previously handled these errors as-is. However the docs I've read in substrate imply that even dropped/usurped etc things may still make it into blocks. If I was to follow that logic, there would be only one error ("finality timeout"), but equally, I imagine that most of the time dropped/usurped/invalid transactions will never again be included.

So, is it best to follow the substrate docs on this, which are super vague about when the transaction ends (finalized, or timed out waiting to finalize after a long period), orr, do I continue to assume that these cases are actually errors and that the transaction is probably scuppered.

I think the latter offhand, but do need to document this properly and will probably ask in dumb questions..

Reply:
^ My current plan; stick with a similar approach to previous but add notes about things and add an issue to revisit

(I added a comment on the timeout!)

Reply:
^ I didn't do that in the end; the higher level APIs now ignore the "error" statuses, knowing that the tx might still be finalized. That felt safer than potentially erroring but your tx still being included!